python/lsst/daf/butler/registry/collections/_base.py

```diff
@@ -323,12 +274,31 @@ def find(self, name: str) -> CollectionRecord[K]:
             raise MissingCollectionError(f"No collection with name '{name}' found.")
         return result
 
+    def _find_many(self, names: Iterable[str]) -> list[CollectionRecord[K]]:
+        """Return multiple records given their names."""
+        names = list(names)
+        # To protect against potential races in cache updates.
```

Comment:
Oof, if this is because FastAPI without async forces threading on us, and we expect the same `SqlRegistry` to be used by multiple threads, there's gonna be a lot more of this, enough so that it makes me think we need to do something else, like have explicit thread-local `SqlRegistry` instance (maybe sharing a connection pool?) or get all caching out of `SqlRegistry`.

Reply:
Well, I was just thinking about concurrency in general, without async context. It's true that caches make everything very difficult, would be better to get rid of caching entirely. We'll have to think about it very soon.